infrastructure/lib/constructs/compliant-resources.ts

```diff
@@ -1,18 +1,29 @@
 import * as s3 from "@aws-cdk/aws-s3";
 import * as cdk from "@aws-cdk/core";
 
-/**
- * Creates a private S3 bucket. (TODO: compliant with NIST SP 800-53 controls)
- */
-export class PrivateBucket extends s3.Bucket {
-  constructor(scope: cdk.Construct, id: string, props?: s3.BucketProps) {
-    super(scope, id, {
-      ...props,
-      // Override any given props with better defaults
+export interface SecureBucketProps {
+  logTargetBucket: s3.IBucket | "self";
+  logTargetPrefix: string;
+  bucketProps?: s3.BucketProps;
+}
+
+export class SecureBucket extends cdk.Construct {
+  public readonly bucket: s3.IBucket;
+
+  constructor(scope: cdk.Construct, id: string, props: SecureBucketProps) {
+    super(scope, id);
+
+    const bucket = new s3.Bucket(this, id, {
+      ...props.bucketProps,
       publicReadAccess: false,
       blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
       autoDeleteObjects: false,
-      // TODO: Implement server access logging, versioning, etc
     });
+    // low level (L1) construct modifications for server access logs
+    (bucket.node.defaultChild as s3.CfnBucket).loggingConfiguration = {
+      logFilePrefix: props.logTargetPrefix,
+      destinationBucketName: props.logTargetBucket === "self" ? undefined : props.logTargetBucket.bucketName,
+    };
```

Comment:
Since we no longer need the reference to the bucket name, this can move directly into the appropriate config on lines 18-20 (directly in the `s3.Bucket` constructor)

Reply:
Moved into the `s3.Bucket` constructor.

Reply:
Fixed in 71f6e4a8971f6a87fc95ac7f65f56550d0de3432